sw/device/lib/dif/dif_edn.c

```diff
@@ -258,13 +258,12 @@ dif_result_t dif_edn_update(const dif_edn_t *edn,
   if (edn == NULL || seed_material == NULL) {
     return kDifBadArg;
   }
-  dif_csrng_seed_material_t seed_material2;
-  memcpy(&seed_material2, seed_material, sizeof(seed_material2));
-  return csrng_send_app_cmd(edn->base_addr, EDN_SW_CMD_REQ_REG_OFFSET,
-                            (csrng_app_cmd_t){
-                                .id = kCsrngAppCmdUpdate,
-                                .seed_material = &seed_material2,
-                            });
+  return csrng_send_app_cmd(
+      edn->base_addr, EDN_SW_CMD_REQ_REG_OFFSET,
+      (csrng_app_cmd_t){
+          .id = kCsrngAppCmdUpdate,
+          .seed_material = (const dif_csrng_seed_material_t *)seed_material,
```

Comment:
I think that the `memcpy` was here to avoid a Strict Aliasing Rule break.
@luismarques could you let us know your thoughts?

Reply:
We can make the structs `dif_edn_seed_material_t` and `dif_csrng_seed_material_t` compatible by making their members have the same name. Also, would it not be neater to just have a common `dif_edn_csrng_seed_material_t` struct that is used by both?

Reply:
Here are the links to the structs:
https://github.com/lowRISC/opentitan/blob/f0ddbd78b0f71a4f2637ef5f84640878ae15321f/sw/device/lib/dif/dif_edn.h#L69
https://github.com/lowRISC/opentitan/blob/f0ddbd78b0f71a4f2637ef5f84640878ae15321f/sw/device/lib/dif/dif_csrng.h#L217